httpx/models.py

```diff
@@ -351,7 +351,12 @@ def update(self, params: QueryParamTypes = None) -> None:  # type: ignore
 
         params = QueryParams(params)
         for param in params:
-            self[param] = params[param]
+            item, *extras = params.getlist(param)
```

Comment:
wondering if there's an scope for refactoring for simplificity here.
Would the following also work?...

```python
items = params.getlist(param)
self._list.extend((param, item) for item in items)
self._dict[param] = items[0]
```

Reply:
I don't think that will work if the params getting replaced are of a list/repeated value.

We need to call `self[param]=` at some point to trigger the search delete for other values in `QueryParams._list` via `QueryParams.__setitem__`

Reply:
Good point!